### PR TITLE
fix(config-migration): fetch raw file from platform, not fs

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Fetch raw config file from platform, not fs. fs may not be initialized.

## Context

This was a mistake during refactoring. Note: we mock detectIntent() so this was not caught/is not tested here, but this fix needs prioritization.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/config-migration/branch/migrated-data.ts [hunk 1]
```diff
@@ -142,10 +143,10 @@ export class MigratedDataFactory {
       delete migratedConfig.errors;
       delete migratedConfig.warnings;
 
-      // indent defaults to 2 spaces
       // TODO #22198
-      const raw = await readLocalFile(configFileName!, 'utf8');
+      const raw = await platform.getRawFile(configFileName!);
       const indent = detectIndent(raw!);
```
Comment (viceice) [inline lib/workers/repository/config-migration/branch/migrated-data.ts:148]:
```suggestion
       const indent = detectIndent(raw ?? '');
```
we should be more defensive here

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Comment by viceice** [comment]
I think tests need a change too?

**Comment by rarkins** [comment]
> I think tests need a change too?

Tests mock the detect-indent package and the raw value is only kept temporarily. We should merge this ASAP so we can fix the regression

**Review by viceice — APPROVED** [review]

**Pull request merged**